keras/utils/training_utils.py

```diff
@@ -5,13 +5,11 @@
 
 
 def _get_available_devices():
-    from tensorflow.python.client import device_lib
-    local_device_protos = device_lib.list_local_devices()
-    return [x.name for x in local_device_protos]
+    return [x.name for x in K.get_session().list_devices()]
 
 
 def _normalize_device_name(name):
-    name = name.lower().replace('device:', '')
+    name = '/' + name.lower().split('device:')[1]
```

Comment:
Why would this line change at all?

Note that tests for this util are not run on Travis, so please run them manually on a multi-GPU machine.

Reply:
The output of the two functions for some reason is different so I need to treat it differently to get the same names out.

Reply:
Why is it different, and what does it look like in both cases?

Reply:
I have no idea what it looks diffferent, this is on the side of Tensorflow. Here is how they look in each case:
```
>>> from tensorflow.python.client import device_lib
>>> l1=[d.name for d in device_lib.list_local_devices()]
>>> l1
[u'/cpu:0']
```
```
>>> from keras import backend as K
>>> l2=[d.name for d in K.get_session().list_devices()]
>>> l2
['/job:localhost/replica:0/task:0/device:CPU:0']
```

That's why I'm splitting on "device:", lowercase and prepend slash to make them look the same which is required in the GPU utils code.

Reply:
BTW this `device_lib.list_local_devices` method is extremely problematic. In earlier versions of TF it was actually registering devices that were not whitelisted in CUDA_VISIBLE_DEVICES. The `K.get_session().list_devices()` method does the trick just fine; I've been using this in large GPU clusters (Keras+Spark) for distributed prediction and never had an issue.